### DISCUSSION TITLE
After using subsetCells(), should AddTileMatrix be rerun?

### DISCUSSION THREAD

**Discussion by macpheke** [GreenleafLab/ArchR#679] [Questions / Documentation]
If I subset my project with subsetCells specifically, do I need to rerun AddTileMatrix and AddGeneScoreMatrix? That is, prior to redoing downstream processes such as iterative LSI, UMAP, MAGIC, etc.

I know it reduces the size of the ArrowFile - presumably to keep info from just the subsetted cells - but I'm wondering if that matters for LSI and the variable features it uses, etc.

The goal here is to subcluster a group of cells to improve resolution.

**Comment by rcorces** [comment]
I think there are a few points of confusion here. Make sure you understand the difference between an ArchRProject and the associated Arrow files.

`subsetCells()` doesnt touch the arrow files, just the ArchRProject.
You are most likely looking for `subsetArchRProject()` - https://www.archrproject.com/reference/subsetArchRProject.html

After `subsetArchRProject()`, you do not need to rerun `addTileMatrix()` or `addGeneScoreMatrix()`

**Reply by macpheke** [reply]
What I mean is, if I use `subsetCells()` on an existing ArchRProject object, then I re-run `addTileMatrix()` and `addGeneScoreMatrix()`, the associated Arrow file is reduced in size. So I am wondering:

- does subsetCells() + addTileMatrix() + addGeneScoreMatrix() perform a similar operation as `subsetArchRProject()` (when `dropCells = TRUE,`) in that it removes inactive cells from the Arrow file?
- would using subsetCells() _without_ re-running addTileMatrix() and addGeneScoreMatrix() adversely affect the accuracy of any subsequent iterative LSI and UMAP?

I ask because `subsetArchRProject()` is not working for me when `dropCells = TRUE` (I'm trying to save space), I get the same error as https://github.com/GreenleafLab/ArchR/issues/663 and a nearly identical logfile but I am using genome hg38. While I try to figure out why I was hoping to use this as a workaround

**Reply by rcorces** [reply]
This isnt the designed use for this function and I would not consider the proposed use as equivalent to `subsetArchRProject()`. Arrow files also contain the fragments and other information that isnt accounted for. I have no idea if you will run into downstream problems using your approach but I would imagine so. Hopefully we have a solution to #663 soon.

**Reply by macpheke** [reply]
Ahh, thank you, I will certainly avoid doing that then.

So in the meanwhile, if I perform `subsetCells()` on an ArchRProject and then re-run `addIterativeLSI()`, `addUMAP()`, `addImputeWeights()` etc.,  the results are still valid even though the associated Arrow file contains unwanted cells/cells not in the desired subset list?

**Reply by rcorces** [reply]
should be! I would think that any downstream functions that run without error are valid.
